meteor/client/ui/Shelf/MiniRundownPanel.tsx

```diff
@@ -0,0 +1,241 @@
+	part: Part | undefined
+}
+
+export class MiniRundownPanelInner extends MeteorReactComponent<
+	IMiniRundownPanelProps & IMiniRundownPanelTrackedProps,
+	IState
+> {
+	constructor(props) {
+		super(props)
+		this.state = {}
+	}
+
+	render() {
+		const isDashboardLayout = RundownLayoutsAPI.isDashboardLayout(this.props.layout)
+		const style = getElementStyle(this.props, isDashboardLayout)
+		const showAny = isShowAnyValues(this.props)
+
+		const current = getMiniRundownPart(
+			this.props,
+			showAny,
+			this.props.currentSegment,
+			this.props.currentPartInstance?.part
+		)
+		const next = getMiniRundownPart(this.props, showAny, this.props.nextSegment, this.props.nextPartInstance?.part)
+
+		const allParts: Part[] = this.props.playlist.getAllOrderedParts()
+		const allSegments: Segment[] = this.props.playlist.getSegments()
+
+		const secondSegmentAndPart = getNextSegmentAndPart(
+			allParts,
+			allSegments,
+			showAny,
+			this.props.nextPartInstance?.part
+		)
+		const second = getMiniRundownPart(this.props, showAny, secondSegmentAndPart.segment, secondSegmentAndPart.part)
+
+		const thirdSegmentAndPart = getNextSegmentAndPart(allParts, allSegments, showAny, secondSegmentAndPart.part)
+		const third = getMiniRundownPart(this.props, showAny, thirdSegmentAndPart.segment, thirdSegmentAndPart.part)
+
+		return (
+			<div
+				className={ClassNames('dashboard-panel mini-rundown-panel', getContainerClass(this.props, isDashboardLayout))}
+				style={getContainerStyle(this.props, isDashboardLayout)}
+			>
+				<span className="mini-rundown-panel__name" style={style}>
+					{showAny && this.props.panel.name}{' '}
+				</span>
+				<div className="current-part">
+					<span className="mini-rundown-panel__rank">{current.identifier}</span>
+					<span className="mini-rundown-panel__segment" style={style}>
+						{current.segmentName}
+					</span>
+					<span className="mini-rundown-panel__part" style={style}>
+						{current.partName}
+					</span>
+				</div>
+
+				<div className="next-part">
+					<span className="mini-rundown-panel__rank">{next.identifier}</span>
+					<span className="mini-rundown-panel__segment" style={style}>
+						{next.segmentName}
+					</span>
+					<span className="mini-rundown-panel__part" style={style}>
+						{next.partName}
+					</span>
+				</div>
+
+				<div className="second-part">
+					<span className="mini-rundown-panel__rank">{second.identifier}</span>
+					<span className="mini-rundown-panel__segment" style={style}>
+						{second.segmentName}
+					</span>
+					<span className="mini-rundown-panel__part" style={style}>
+						{second.partName}
+					</span>
+				</div>
+
+				<div className="third-part">
+					<span className="mini-rundown-panel__rank">{third.identifier}</span>
+					<span className="mini-rundown-panel__segment" style={style}>
+						{third.segmentName}
+					</span>
+					<span className="mini-rundown-panel__part" style={style}>
+						{third.partName}
+					</span>
+				</div>
```

Comment:
According to the task (unless it is outdated, or I misunderstand it), the list should be scrollable by the user. It should show all the segments, but make sure that when Next changes (due to user actions, including tapping the buttons to the left etc.), it is scrolled to be the 2nd position on the list.

> 5. Verify that it’s possible to navigate through the segments for view only
> 7. If the entry of the list which renders as Green has an index > 0 (not the first entry in the list), then the list should scroll to show the Green entry in the second position of the list. The list will keep scrolling and make sure that the Green one stays in position 2 within the viewport every time it updates (when Next updates (as a cosequence of Takes etc.))

Reply:
You might be right. From the description and the markup I understood it as a static list that would update on takes